unit_tests/testVTUFiles/testVTUFiles.f90

```diff
@@ -49,92 +50,118 @@ SUBROUTINE SetupError
-      bool = .NOT.testVTUFile%hasMesh
-      ASSERT(bool, 'testVTUFile%writeMesh(...) STRUCTURED_POINTS')
+      bool=.NOT.testVTUFile%hasMesh
+      ASSERT(bool,'testVTUFile%writeMesh(...) STRUCTURED_POINTS')
       CALL testVTUFile%clear(.TRUE.)
     ENDSUBROUTINE testFailSTRUCTURED_POINTS
 !
 !-------------------------------------------------------------------------------
-    SUBROUTINE testUNSTRUCTURED_GRID
+    SUBROUTINE testUNSTRUCTURED_GRID()
       !Test writeMesh for UNSTRUCTURED_GRID
       CALL testVTUFile%initialize(666,'testVTU2.vtu')
-      CALL SetupTest2_Mesh
+      CALL SetupTest2_Mesh()
       CALL testVTUFile%writeMesh(testVTKMesh)
-      bool = testVTUFile%hasMesh
-      ASSERT(bool, 'testVTUFile%writeMesh(...) UNSTRUCTURED_GRID')
-
+      bool=testVTUFile%hasMesh
+      ASSERT(bool,'testVTUFile%writeMesh(...) UNSTRUCTURED_GRID')
+      !
       !Write data on the mesh
-      CALL SetupTest2_Data
+      CALL SetupTest2_Data()
       CALL testVTUFile%writeScalarData(testVTKData)
-      testVTKData%varname='mat_val'
+      !
+      !Test writeScalarData
+      COMPONENT_TEST('write float')
       testVTKData%vtkDataFormat='float'
+      testVTKData%varname='mat_float'
       CALL testVTUFile%writeScalarData(testVTKData)
-
-      !Test writeScalarData
+      CALL testVTUFile%hasData('mat_float','float',bool)
+      ASSERT(bool,'testVTUFile%hasData(...) FLOAT')
+      !
+      COMPONENT_TEST('write long')
       testVTKData%vtkDataFormat='long'
       testVTKData%varname='mat_long'
       CALL testVTUFile%writeScalarData(testVTKData) !repeat for long
+      CALL testVTUFile%hasData('mat_long','long',bool)
+      ASSERT(bool,'testVTUFile%hasData(...) LONG')
+      !
+      COMPONENT_TEST('write short')
       testVTKData%vtkDataFormat='short'
       testVTKData%varname='mat_short'
       CALL testVTUFile%writeScalarData(testVTKData) !repeat for short
+      CALL testVTUFile%hasData('mat_short','short',bool)
+      ASSERT(bool,'testVTUFile%hasData(...) SHORT')
+      !
+      COMPONENT_TEST('write int')
       testVTKData%vtkDataFormat='int'
-      testVTKData%varname='mat_float'
+      testVTKData%varname='mat_int'
       CALL testVTUFile%writeScalarData(testVTKData) !write as int
+      CALL testVTUFile%hasData('mat_int','int',bool)
+      ASSERT(bool,'testVTUFile%hasData(...) INT')
+      !
+      COMPONENT_TEST('write double')
       testVTKData%vtkDataFormat='double'
       testVTKData%varname='mat_double'
       CALL testVTUFile%writeScalarData(testVTKData) !write as double
-      CALL testVTUFile%close
-      CALL testVTUFile%clear
+      CALL testVTUFile%hasData('mat_double','double',bool)
+      ASSERT(bool,'testVTUFile%hasData(...) DOUBLE')
+      !
+      CALL testVTUFile%close()
+      CALL testVTUFile%clear()
     ENDSUBROUTINE testUNSTRUCTURED_GRID
 !
 !-------------------------------------------------------------------------------
-    SUBROUTINE testMultiVTU
+    SUBROUTINE testMultiVTU()
```

Comment:
This routine needs some ASSERTS as well. If you're unsure about this, the question to ask yourself is "what is the clearest and most direct way to assert that what I am testing works?"